catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/MetacardVersion.java

```diff
@@ -196,6 +284,18 @@ public MetacardVersion(Metacard sourceMetacard, Action action, Subject subject)
         this.setVersionOfId(sourceMetacard.getId());
         this.setVersionTags(sourceMetacard.getTags());
 
+        Optional<MetacardType> type = types.stream()
```

Comment:
This block of code is currently duplicated in this file and is a good target for extraction to a method.